lab-yunjoo/test/note-route-test.js

```diff
@@ -0,0 +1,156 @@
+'use strict';
+
+const request = require('superagent');
+const expect = require('chai').expect;
+const server = require('./../server');
+const port = process.env.PORT || 3000;
+const serverUrl = `http://localhost:${port}`;
+
+describe('testing note-route module', function(){
+  before(function(done){
+    if(!server.isRunning){
+      server.listen(port, function(){
+        server.isRunning = true;
+        done();
+      });
+      return;
+    }
+    done();
+  });
+  after(function(done){
+    if(server.isRunning){
+      return server.close(() => {
+        server.isRunning = false;
+        done();
+      });
+    }
+    return;
+  });
+
+});
+describe('tesing bad POST request on endpoint /api/note', function(){
+  it('should return not found status 404 code',(done)=>{
+    request.post(`${serverUrl}/api/n`)
+      .send({content:'meh~'})
+      .end((err,res) =>{
+        this.res = res;
+        this.note = res.body;
+        expect(this.res.status).to.equal(404);
+        done();
+      });
+  });
+  it('should return bad request 400',(done) =>{
+    request.post(`${serverUrl}/api/note`)
+      .send({})
+      .end((err,res) =>{
+        this.res = res;
```

Comment:
Since you aren't breaking your requests out into before blocks, and then running your expects in it blocks, there isn't a point to doing all the 'this.res = res' stuff. You can just skip ahead to something like 
```javascript
expect(res.status).to.equal(404);
```